add checklist (#66)
nrfconnect/ncs-app-index: docs/ContributionChecklist.md

> # Addon Contributors' Checklist

Reviewer: ```suggestion
# Add-on contributor checklist
```

---

> The following criteria need to be met in order to contribute a new addon.

Reviewer: ```suggestion
You must meet the following criteria if you want to contribute with a new add-on:
```

> - [ ] Register a new account on [Nordic Semiconductor Devzone](https://devzone.nordicsemi.com/) using your company's e-mail provided as contact 
> in our mailing list.

Reviewer: ```suggestion
- [ ] Register a new account on [Nordic Semiconductor DevZone](https://devzone.nordicsemi.com/) using your company's e-mail provided as contact in our mailing list.
```

> - [ ] Create a Pull Request in the [NCS Addons repository](https://github.com/nrfconnect/ncs-app-index). This Pull Request should introduce a new file describing your organization and the addons. It should also include the proof of reliability of the new addons, for example a brief description of tests executed. More details can be found in [Contributing](https://github.com/nrfconnect/ncs-app-index/blob/main/CONTRIBUTING.md).

Reviewer: ```suggestion
- [ ] Create a Pull Request to the [nRF Connect SDK Add-on repository](https://github.com/nrfconnect/ncs-app-index). This Pull Request should add the new file that describes your organization and the add-ons you are adding. It should also include the proof of reliability of the new add-ons, for example a brief description of tests executed. More details can be found in [Contributing](https://github.com/nrfconnect/ncs-app-index/blob/main/CONTRIBUTING.md).
```

> - [ ] The Pull Request will be approved and merged once CI/CD passes.

Reviewer: ```suggestion
- [ ] Make sure the CI/CD passes for the Pull Request to be approved and merged.
```
